src/goSuggest.ts

```diff
@@ -199,7 +199,17 @@ export class GoCompletionItemProvider implements vscode.CompletionItemProvider {
 										paramSnippets.push('${' + (i + 1) + ':' + param + '}');
 									}
 								}
-								item.insertText = new vscode.SnippetString(suggest.name + '(' + paramSnippets.join(', ') + ')');
+								let currentWordIndex = lineText.toLowerCase().lastIndexOf(currentWord.toLowerCase());
+								let nextSymbolIndex = currentWordIndex + currentWord.length;
+								let newSnippetString = null;
+								// Avoid adding snippet for function suggest when cursor is followed by ()
+								// i.e: met() -> method()()
+								if ((nextSymbolIndex < lineText.length) && (lineText.charAt(nextSymbolIndex) === '(')) {
```

Comment:
Why not check `lineText.substr(position.character, 2) === '()'` instead?

Reply:
Done